torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1901,6 +1909,26 @@ def test_remote_throw(self):
         with self.assertRaisesRegex(Exception, ".*Expected error.*"):
             rref.to_here()
 
+    @dist_init
+    def test_non_cont_tensors(self):
+        if self.rank == 0:
+            # Create a non-contiguous tensor.
+            t = torch.rand(5, 5)
+            t_view = t.narrow(1, 2, 2)
+            self.assertFalse(t_view.is_contiguous())
+            t_cont = t_view.contiguous()
+            self.assertTrue(t_cont.is_contiguous())
+            self.assertEqual(t_view, t_cont)
+
+            # Send non-cont tensor over RPC.
+            next_rank = (self.rank + 1) % self.world_size
+            t_ret = rpc.rpc_sync(worker_name(next_rank), non_cont_test, args=(t_view, t_cont))
```

Comment:
Isn't this behavior depends on the sparsity of the data?

https://github.com/pytorch/pytorch/blob/9e016f77a8fc8c628f8a80f594cd071ae48ddf21/torch/csrc/distributed/rpc/utils.cpp#L236

Reply:
Yes, although given the small size of the tensor here I don't think that we're copying the tensor. This test was mostly to ensure we can send non-contiguous tensors over the wire without any correctness issues.